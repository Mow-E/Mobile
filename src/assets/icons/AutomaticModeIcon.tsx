import * as React from 'react';
import Svg, {SvgProps, Path} from 'react-native-svg';
import IconProps from './IconProps';
import colors from '../../styles/colors';

function SvgComponent(props: SvgProps): JSX.Element {
  return (
    <Svg width={24} height={24} viewBox="0 0 22 24" fill="none" {...props}>
      <Path
        d="M4.52273 24H0.795455L9.17045 0.727272H13.2273L21.6023 24H17.875L11.2955 4.95454H11.1136L4.52273 24ZM5.14773 14.8864H17.2386V17.8409H5.14773V14.8864Z"
        fill={props.stroke}
      />
    </Svg>
  );
}

/**
 * An icon for the automatic mower mode.
 */
function AutomaticModeIcon({
  size = 24,
  darkModeInverted = false,
}: IconProps): JSX.Element {
  const strokeColor = darkModeInverted ? colors.gray['50'] : colors.gray['950'];

  return <SvgComponent height={size} width={size} stroke={strokeColor} />;
}

export default AutomaticModeIcon;
