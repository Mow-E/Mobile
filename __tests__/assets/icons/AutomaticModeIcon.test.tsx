import 'react-native';
import React from 'react';
import AutomaticModeIcon from '../../../src/assets/icons/AutomaticModeIcon';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders default correctly', () => {
  const tree = renderer.create(<AutomaticModeIcon />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders colored correctly', () => {
  const tree = renderer.create(<AutomaticModeIcon colored />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders uncolored correctly', () => {
  const tree = renderer.create(<AutomaticModeIcon colored={false} />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders in dark mode correctly', () => {
  const tree = renderer.create(<AutomaticModeIcon darkModeInverted />).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders in different sizes', () => {
  const treeSmall = renderer.create(<AutomaticModeIcon size={24} />).toJSON();
  const treeLarge = renderer.create(<AutomaticModeIcon size={48} />).toJSON();

  expect(treeSmall).not.toEqual(treeLarge);
});
