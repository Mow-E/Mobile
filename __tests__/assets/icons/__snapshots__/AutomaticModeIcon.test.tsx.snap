// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders colored correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={22}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGPath
      d="M4.52273 24H0.795455L9.17045 0.727272H13.2273L21.6023 24H17.875L11.2955 4.95454H11.1136L4.52273 24ZM5.14773 14.8864H17.2386V17.8409H5.14773V14.8864Z"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders default correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={22}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGPath
      d="M4.52273 24H0.795455L9.17045 0.727272H13.2273L21.6023 24H17.875L11.2955 4.95454H11.1136L4.52273 24ZM5.14773 14.8864H17.2386V17.8409H5.14773V14.8864Z"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders in dark mode correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#ffffff"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={22}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4294967295,
        "type": 0,
      }
    }
  >
    <RNSVGPath
      d="M4.52273 24H0.795455L9.17045 0.727272H13.2273L21.6023 24H17.875L11.2955 4.95454H11.1136L4.52273 24ZM5.14773 14.8864H17.2386V17.8409H5.14773V14.8864Z"
      fill={
        {
          "payload": 4294967295,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;

exports[`renders uncolored correctly 1`] = `
<RNSVGSvgView
  align="xMidYMid"
  bbHeight="24"
  bbWidth="24"
  fill="none"
  focusable={false}
  height={24}
  meetOrSlice={0}
  minX={0}
  minY={0}
  stroke="#000000"
  style={
    [
      {
        "backgroundColor": "transparent",
        "borderWidth": 0,
      },
      {
        "flex": 0,
        "height": 24,
        "width": 24,
      },
    ]
  }
  vbHeight={24}
  vbWidth={22}
  width={24}
>
  <RNSVGGroup
    fill={null}
    propList={
      [
        "fill",
        "stroke",
      ]
    }
    stroke={
      {
        "payload": 4278190080,
        "type": 0,
      }
    }
  >
    <RNSVGPath
      d="M4.52273 24H0.795455L9.17045 0.727272H13.2273L21.6023 24H17.875L11.2955 4.95454H11.1136L4.52273 24ZM5.14773 14.8864H17.2386V17.8409H5.14773V14.8864Z"
      fill={
        {
          "payload": 4278190080,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    />
  </RNSVGGroup>
</RNSVGSvgView>
`;
